private/bufpkg/bufplugin/bufpluginconfig/get.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufpluginconfig
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/bufbuild/buf/private/pkg/encoding"
+	"github.com/bufbuild/buf/private/pkg/storage"
+	"github.com/bufbuild/buf/private/pkg/stringutil"
+	"go.opencensus.io/trace"
+	"go.uber.org/multierr"
+)
+
+func getConfigForBucket(ctx context.Context, readBucket storage.ReadBucket) (_ *Config, retErr error) {
```

Comment:
Will `getConfigForBucket` work the same as `buf.yaml`, i.e. are we expecting to pass a bucket where `buf.plugin.yaml` should be at the root? I guess so yea